src/pages/home/report/withReportAndPrivateNotesOrNotFound.js

```diff
@@ -0,0 +1,129 @@
+import getComponentDisplayName from '../../../libs/getComponentDisplayName';
+import reportPropTypes from '../../reportPropTypes';
+import withReportOrNotFound from './withReportOrNotFound';
+import * as Report from '../../../libs/actions/Report';
+import NotFoundPage from '../../ErrorPage/NotFoundPage';
+import FullScreenLoadingIndicator from '../../../components/FullscreenLoadingIndicator';
+import {withNetwork} from '../../../components/OnyxProvider';
+import compose from '../../../libs/compose';
+import * as ReportUtils from '../../../libs/ReportUtils';
+import networkPropTypes from '../../../components/networkPropTypes';
+import ONYXKEYS from '../../../ONYXKEYS';
+
+export default function (WrappedComponent) {
+    const propTypes = {
+        /** The HOC takes an optional ref as a prop and passes it as a ref to the wrapped component.
+         * That way, if a ref is passed to a component wrapped in the HOC, the ref is a reference to the wrapped component, not the HOC. */
+        forwardedRef: PropTypes.func,
+
+        /** The report currently being looked at */
+        report: reportPropTypes,
+
+        /** Information about the network */
+        network: networkPropTypes.isRequired,
+
+        /** Session of currently logged in user */
+        session: PropTypes.shape({
+            /** Currently logged in user accountID */
+            accountID: PropTypes.number,
+        }),
+
+        route: PropTypes.shape({
+            /** Params from the URL path */
+            params: PropTypes.shape({
+                /** reportID and accountID passed via route: /r/:reportID/notes */
+                reportID: PropTypes.string,
+                accountID: PropTypes.string,
+            }),
+        }).isRequired,
+    };
+
+    const defaultProps = {
+        forwardedRef: () => {},
+        report: {},
+        session: {
+            accountID: null,
+        },
+    };
+
+    // eslint-disable-next-line rulesdir/no-negated-variables
+    function WithReportAndPrivateNotesOrNotFound(props) {
+        const {route, report, network, session} = props;
+        const accountID = route.params.accountID;
+        const isLoadingPrivateNotes = report.isLoadingPrivateNotes;
+
+        useEffect(() => {
+            if (network.isOffline && report.isLoadingPrivateNotes) {
+                return;
+            }
+
+            Report.getReportPrivateNote(report.reportID);
+            // eslint-disable-next-line react-hooks/exhaustive-deps -- do not add isLoadingPrivateNotes to dependencies
+        }, [report.reportID, network.isOffline]);
+
+        const isPrivateNotesEmpty = accountID ? _.isEmpty(lodashGet(report, ['privateNotes', accountID, 'note'], '')) : _.isEmpty(report.privateNotes);
+        const shouldShowFullScreenLoadingIndicator = isLoadingPrivateNotes !== false && isPrivateNotesEmpty;
+
+        // eslint-disable-next-line rulesdir/no-negated-variables
+        const shouldShowNotFoundPage = useMemo(() => {
```

Comment:
Why this value is memoized?

Reply:
I am calling `ReportUtils.isArchivedRoom(report)`. In addition, I am doing conditional checks in the memoized function. I dont want this to be re-computed every time until deps change.